cql3/functions/functions.cc

```diff
@@ -126,17 +122,19 @@ void functions::add_function(shared_ptr<function> func) {
 
 template <typename F>
 void functions::with_udf_iter(const function_name& name, const std::vector<data_type>& arg_types, F&& f) {
-    auto i = find_iter(name, arg_types);
-    if (i == _declared.end() || i->second->is_native()) {
+    auto cit = find_iter(name, arg_types);
+    if (cit == _declared.end() || cit->second->is_native()) {
         log.error("attempted to remove or alter non existent user defined function {}({})", name, arg_types);
         return;
     }
-    f(i);
+    // erase here is only to convert from const_iterator to iterator
+    auto it = _declared.erase(cit, cit);
+    f(it);
 }
 
 void functions::replace_function(shared_ptr<function> func) {
-    with_udf_iter(func->name(), func->arg_types(), [func] (functions::declared_t::iterator i) {
-        i->second = std::move(func);
+    with_udf_iter(func->name(), func->arg_types(), [func] (auto it) {
```

Comment:
auto parameters reduce readability. To guess the type the reader has to inspect the implementation of with_udf_iter.

Reply:
also, this turns this parameter into a template parameter, so this lambda is now a templated (or generic) lambda, if we really care about compiling speed, might take this into consideration and probably want to avoid this.

Reply:
fixed